client/src/components/Overview/ProductSummary.jsx

```diff
@@ -1,76 +1,38 @@
 import React from 'react';
 import axios from 'axios';
 import AtelierApi from '../../lib/atelierAPI.js';
+import { useSelector } from 'react-redux';
 
-export default class ProductSummary extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      product: {},
-      reviewMetadata: {},
-      starPercentage: '',
-      count: ''
-    };
-  }
-
-  componentDidMount() {
-    const productId = 16060;
-    this.renderProductInfo(productId);
-    this.renderReviewInfo(productId);
-  }
-
-  renderProductInfo(productId) {
-    axios.get(`${AtelierApi.url}/products/${productId}`, {
-      headers: AtelierApi.headers
-    })
-      .then(response => {
-        this.setState({
-          product: response.data
-        });
-      })
-      .catch(err => console.log(err));
-  }
+const ProductSummary = () => {
+  const productInfo = useSelector((state) => state.product.productInfo);
 
-  renderReviewInfo(productId) {
-    axios.get(`${AtelierApi.url}/reviews/meta`, {
-      // eslint-disable-next-line camelcase
-      params: { product_id: productId },
-      headers: AtelierApi.headers
-    })
-      .then(response => {
-        this.setState({
-          reviewMetadata: response.data
-        });
-      })
-      .then(() => {
-        let ratings = 0;
-        let count = 0;
-        const eachRating = this.state.reviewMetadata.ratings;
-        for (var key in eachRating) {
-          count += Number(eachRating[key]);
-          ratings += (Number(key) * Number(eachRating[key]));
-        }
-        let starPercentage = (ratings / count) / 5 * 100;
-        // console.log(starPercentage);
-        if ((starPercentage % 5) < 2.5) {
-          starPercentage -= (starPercentage % 5);
-        } else {
-          starPercentage += (5 - (starPercentage % 5));
-        }
-
-        this.setState({
-          count: count,
-          starPercentage: starPercentage
-        }, () => {
-          // console.log(this.state.starPercentage);
-          document.querySelector('#body-overview-star .stars-inner').style.width = `${this.state.starPercentage}%`;
-        });
-
-      })
-      .catch(err => console.log(err));
-  }
+  const styleInfo = useSelector(state => {
+    if (state.product.styleInfo.results === undefined) {
+      return undefined;
+    }
+    return {
+      price: state.product.styleInfo.results[state.style]
+    };
+  });
 
-  render() {
+  console.log(styleInfo);
+  if (styleInfo && !styleInfo.price.sale_price) {
```

Comment:
You could combine this call to useSelector with the next call to usesSelector:

```suggestion
  const state = useSelector(state => {
    if (state.product.styleInfo.results === undefined) {
      return undefined;
    }
    return {
      productInfo: state.product.productInfo,
      price: state.product.styleInfo.results[state.style].price
    };
  });
  
  if (!state) {
    return (<div>Loading...</div>);
  }
  return (
    ...
```

Then your calls later would be simplified to e.g., price.original_price instead of styleInfo.price.original_price.

Reply:
In this case, I still need state.price to specify the key in state that I need.

Reply:
Line 15 of the suggested change pulls out the price info for use in the other parts of your code.